cmd/api/src/database/sso_providers.go

```diff
@@ -104,3 +116,12 @@ func (s *BloodhoundDB) DeleteSSOProvider(ctx context.Context, id int) error {
 
 	return err
 }
+
+// GetSSOProviderUsers returns all the users associated with a given sso provider
+func (s *BloodhoundDB) GetSSOProviderUsers(ctx context.Context, id int) (model.Users, error) {
+	var (
+		users model.Users
+	)
+
+	return users, CheckError(s.db.WithContext(ctx).Table("users").Where("sso_provider_id = ?", id).Find(&users))
+}
```

Comment:
Yeah, I had the same thinking.

Since it returns `model.Users` I was thinking it makes more sense to live alongside the queries pertaining to `Users`, but in this instance I followed what the SAML implementation did:
https://github.com/SpecterOps/BloodHound/blob/3681d66c3bff4c6bd0d4295f6af00b8c485dda53/cmd/api/src/database/db.go#L133

Reply:
Fair enough 🤷